moshi/build.gradle.kts

```diff
@@ -22,11 +22,43 @@ plugins {
   id("ru.vyarus.animalsniffer")
 }
 
+val mainSourceSet by sourceSets.named("main")
+val java16 by sourceSets.creating {
+  java {
+    srcDir("src/main/java16")
+  }
+}
+
+tasks.named<JavaCompile>("compileJava16Java") {
+  javaCompiler.set(
+    javaToolchains.compilerFor {
+      languageVersion.set(JavaLanguageVersion.of(16))
+    }
+  )
+  options.release.set(16)
+}
+
+// We need to include our actual RecordJsonAdapter from java16 sources and replace the shime
+tasks.named<Jar>("jar") {
+  from(java16.output) {
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
+  }
+  manifest {
+    // Indicate that this is a multi release jar
```

Comment:
Is it? Just looks like a regular jar to me.

Reply:
I was sort of assuming this was needed if the jar contains classes targeting multiple java versions but can omit it if it doesn't really change anything

Reply:
3eb768f